extensions/standard-processors/tests/unit/SplitJsonTests.cpp

```diff
@@ -0,0 +1,150 @@
+    REQUIRE(split_json_processor_);
+    LogTestController::getInstance().setTrace<processors::SplitJson>();
+  }
+
+ protected:
+  void verifySuccessfulSplit(const std::string& input_json_content, const std::string& json_path_expression, const std::vector<std::string>& expected_split_contents) {
+    REQUIRE(controller_.plan->setProperty(split_json_processor_, processors::SplitJson::JsonPathExpression, json_path_expression));
+
+    auto result = controller_.trigger({{.content = input_json_content, .attributes = {{"filename", "unique_file_name"}}}});
+
+    CHECK(result.at(processors::SplitJson::Failure).empty());
+    REQUIRE(result.at(processors::SplitJson::Original).size() == 1);
+    auto original_flow_file = result.at(processors::SplitJson::Original).at(0);
+    CHECK(controller_.plan->getContent(original_flow_file) == input_json_content);
+    auto original_fragment_id = original_flow_file->getAttribute(processors::SplitJson::FragmentIdentifier.name);
+    CHECK_FALSE(original_fragment_id->empty());
+    auto fragment_count = original_flow_file->getAttribute(processors::SplitJson::FragmentCount.name);
+    CHECK(fragment_count.value() == std::to_string(expected_split_contents.size()));
+
+    REQUIRE(result.at(processors::SplitJson::Split).size() == expected_split_contents.size());
+    for (size_t i = 0; i < expected_split_contents.size(); ++i) {
+      auto flow_file = result.at(processors::SplitJson::Split).at(i);
+      CHECK(controller_.plan->getContent(flow_file) == expected_split_contents[i]);
+      auto fragment_id = flow_file->getAttribute(processors::SplitJson::FragmentIdentifier.name);
+      CHECK(fragment_id.value() == original_fragment_id.value());
+      CHECK(flow_file->getAttribute(processors::SplitJson::FragmentCount.name).value() == std::to_string(expected_split_contents.size()));
+      CHECK(flow_file->getAttribute(processors::SplitJson::FragmentIndex.name).value() == std::to_string(i));
+      CHECK(flow_file->getAttribute(processors::SplitJson::SegmentOriginalFilename.name).value() == "unique_file_name");
+      CHECK_FALSE(flow_file->getAttribute(core::SpecialFlowAttribute::FILENAME).value() == "unique_file_name");
+    }
+  }
+
+  SingleProcessorTestController controller_;
+  core::Processor* split_json_processor_;
+};
+
+TEST_CASE_METHOD(SplitJsonTestFixture, "Query fails with parsing issues", "[SplitJsonTests]") {
+  ProcessorTriggerResult result;
+  std::string error_log;
+  REQUIRE(controller_.plan->setProperty(split_json_processor_, processors::SplitJson::JsonPathExpression, "invalid json path"));
+  SECTION("Flow file content is empty") {
+    result = controller_.trigger({{.content = ""}});
+    error_log = "FlowFile content is empty, transferring to Failure relationship";
+  }
+
+  SECTION("Flow file content is invalid json") {
+    result = controller_.trigger({{.content = "invalid json"}});
+    error_log = "FlowFile content is not a valid JSON document, transferring to Failure relationship";
```

Comment:
```suggestion
    error_log = "FlowFile content is not a valid JSON document, transferring to the 'failure' relationship";
```

Reply:
Updated in https://github.com/apache/nifi-minifi-cpp/pull/1980/commits/0aa867f69a964c3d3f0a1562750bd81218fc6434